finetune/base.py

```diff
@@ -245,6 +246,42 @@ def _distribute_strategy(self, visible_gpus):
         self.resolved_gpus = resolved_gpus
         return distribute_strategy
 
+
+    def load_separate(self,path):
+        """
+        Initialize a model with target and featurizer split into two estimators. This allows
+        faster loading and changing weights within a constant architecture in deployment.
+        """
+        assert self.config.build_separate_estimators,\
+        "Must have 'build separate estimators' enabled in config for post-initialization loading."
+        assert self.config.bert_adapter_size is not None,\
+        "Must have adapters turned on in config to keep loaded file size low"
+        self._load_path = path
+        orig = self.saver.load(path)
+        self.need_to_refresh = True
+        self.input_pipeline = orig.input_pipeline
+        self.get_separate_estimators(path)
+
+    def change_weights(self,path):
+        """
+        Initialize a model with target and featurizer split into two estimators. This allows
+        faster loading and changing weights within a constant neural architecture in deployment.
+        """
+        assert hasattr(self,'featurizer_est') and hasattr(self,'target_est'), "You must call \
```

Comment:
This would be better to keep these defined as None and then test if they are None here. Also, if this is in the public API this should not be an assertion but a FinetuneError or other relevant exception. Assertions are designed to enforce things we believe to be true. Not as routine error handling.